tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/impl/BlobStoreBackedInputStreamImpl.java

```diff
@@ -115,7 +115,8 @@ public int read(byte[] b, int off, int len) throws IOException {
     }
 
     @Override
-    public void seek(long position) {
+    public void seek(long position) throws IOException {
+        refillBufferIfNeeded();
```

Comment:
Because we add a new line in the `BlobStoreBackedHandleImpl`:
```
                inputStream.seek(index.getIndexEntryForEntry(firstEntry).getDataOffset());
```

Before this, the readAsync will use readInt() to trigger the stream to fill buffer, then to seek. Currently, we will execute seek before any read operation. We need to fill the buffer to let it fetch the data.

Reply:
The `position` is the position of the whole object, and the buffer is part of the object. I think we don't need to check `buffer.readerIndex() == position`, the seek operation is not a complicated work.

Reply:
It was added because my test case has some issues. Remove this and refactor the test case